src/legacy/server/i18n/index.js

```diff
@@ -20,49 +20,66 @@
 import { resolve } from 'path';
 import globby from 'globby';
 import { i18n, i18nLoader } from '@kbn/i18n';
-
+import { readFile } from 'fs';
 import { fromRoot } from '../../utils';
+import { promisify } from 'util';
+
+const readFileAsync = promisify(readFile);
+
+export async function getTranslationPaths({ cwd, glob }) {
+  const entries = await globby(glob, { cwd });
+
+  const configFiles = await Promise.all(
+    entries
+      .map(entry => resolve(cwd, entry))
+      .map(async configFilePath => ({
+        path: configFilePath,
+        content: await readFileAsync(configFilePath, 'utf8')
+      }))
+  );
+
+  return configFiles
+    .map(({ content, path }) => {
+      try {
+        return JSON.parse(content);
+      } catch (err) {
+        throw new Error(`Failed to parse .i18nrc.json file at ${path}`);
+      }
+    })
+    .reduce((acc, configFile) => acc.concat(...configFile.translations), [])
+    .map(translationPath => resolve(cwd, translationPath));
+}
 
 export async function i18nMixin(kbnServer, server, config) {
   const locale = config.get('i18n.locale');
+  const localeRegExp = new RegExp(`${locale}.json$`);
```

Comment:
I'm searching all files that end with the file pattern. The start is a full folder path not just the filename (ie not `en.json` but `.../kibana/x-pack/../en.json`)

Reply:
Hmm, I see, then you should make sure that slash is included into regex or something like this, so that if locale is `en` you won't match something like `hack_me_en.json`. If it works as I expect, then it's fine, it just wasn't immediately clear from looking at this regex alone.

Reply:
I agree 👍